x-pack/solutions/chat/plugins/workchat-app/public/application/components/chat/chat_conversation_progression.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { css, keyframes } from '@emotion/react';
+import { EuiTimelineItem, EuiLoadingElastic, EuiAvatar } from '@elastic/eui';
+import type { ProgressionEvent } from '../../../../common/chat_events';
+import type { ProgressionConversationItem } from '../../utils/conversation_items';
+
+interface ChatConversationMessageProps {
+  progress: ProgressionConversationItem;
+}
+
+export const ChatConversationProgression: React.FC<ChatConversationMessageProps> = ({
+  progress,
+}) => {
+  const { progressionEvents } = progress;
+
+  const getText = (event: ProgressionEvent): string => {
+    switch (event.data.step) {
+      case 'planning':
+        return 'Thinking about which integration to use';
+      case 'retrieval':
+        return 'Calling relevant integrations';
+      case 'analysis':
+        return 'Analysing the results';
+      case 'generate_summary':
+        return 'Summarizing content';
+      default:
+        return 'Working';
+    }
+  };
```

Comment:
Hardcoded for now... will of course have to be adapted to be more generic later (the labels will likely come from the server)